controllers/bootstrap/init.go

```diff
@@ -438,8 +438,37 @@ func (b *Bootstrap) CreateOrUpdateFromYaml(yamlContent []byte, alwaysUpdate ...b
 		if len(alwaysUpdate) != 0 {
 			forceUpdate = alwaysUpdate[0]
 		}
-		// TODO: deep merge and update
-		if compareVersion(obj.GetAnnotations()["version"], objInCluster.GetAnnotations()["version"]) || forceUpdate {
+		update := forceUpdate
+
+		// do not compareVersion if the resource is subscription
+		if gvk.Kind == "Subscription" {
+			sub := b.GetSubscription(ctx, obj.GetName(), b.CSData.MasterNs)
+
+			var opt util.OperatorSub
+			opt.Channel = util.GetNestedString(obj, "spec", "channel").(string)
+			opt.InstallPlanApproval = util.GetNestedString(obj, "spec", "installPlanApproval").(string)
+			opt.Name = util.GetNestedString(obj, "spec", "name").(string)
+			opt.Source = util.GetNestedString(obj, "spec", "source").(string)
+			opt.SourceNamespace = util.GetNestedString(obj, "spec", "sourceNamespace").(string)
+			optEnv := util.GetNestedSlice(obj, "spec", "config", "env")
+			var optEnvSlice []map[string]interface{}
+			for _, s := range optEnv {
+				optEnvSlice = append(optEnvSlice, s.(map[string]interface{}))
+			}
+			if len(optEnvSlice) > 0 {
+				opt.InstallScope = optEnvSlice[0]["value"].(string)
+				opt.IsolatedMode = optEnvSlice[1]["value"].(string)
```

Comment:
It is good for now, but if the length of optEnvSlice is 1, then we will get out of range issue.

Reply:
Good catch! but `optEnvSlice` is from the subscription template. for now, there are either 2 items or none.
I can add an extra length check or add a TODO comment for the future if the template is changed. 
which way do we prefer?

Reply:
I would prefer the first one. Because in subscription of NSS operator, we do not have those kind of Config Variables.